### ISSUE TITLE
 ERROR- /home/featurize/work/mmdetection/mmdet/evaluation/metrics/coco metric.py - compute metrics。The testing results of the whole dataset is empty.

### ISSUE THREAD

**Issue by RDXiaoLu** [open-mmlab/mmyolo#790]
### Prerequisite

- [X] I have searched [the existing and past issues](https://github.com/open-mmlab/mmyolo/issues) but cannot get the expected help.
- [X] I have read the [FAQ documentation](https://mmyolo.readthedocs.io/en/latest/faq.html) but cannot get the expected help.
- [X] The bug has not been fixed in the [latest version](https://github.com/open-mmlab/mmyolo).

### 🐞 Describe the bug

使用和mmyolo cat COCO数据一样的 自己制作的数据集。使用yolov8 但是loss_bbox 和 loss_df1为0。
使用yolox是一样的现象。

<img width="734" alt="f48bf361e26c4ee5d60236c5d3b4e1b" src="https://github.com/open-mmlab/mmyolo/assets/87572685/a9aabf39-9206-4517-81b8-9af55c9d9fab">

使用生成的pth推理测试集，在showdir查看，发现无标注框，也无bbox。

### Environment

OrderedDict([('sys.platform', 'linux'),
             ('Python', '3.7.10 (default, Jun  4 2021, 14:48:32) [GCC 7.5.0]'),
             ('CUDA available', True),
             ('numpy_random_seed', 2147483648),
             ('GPU 0', 'NVIDIA GeForce RTX 3060'),
             ('CUDA_HOME', '/usr/local/cuda'),
             ('NVCC', 'Cuda compilation tools, release 11.2, V11.2.152'),
             ('GCC', 'gcc (Ubuntu 9.3.0-17ubuntu1~20.04) 9.3.0'),
             ('PyTorch', '1.10.0+cu113'),
             ('PyTorch compiling details',
              'PyTorch built with:\n  - GCC 7.3\n  - C++ Version: 201402\n  - Intel(R) Math Kernel Library Version 2020.0.0 Product Build 20191122 for Intel(R) 64 architecture applications\n  - Intel(R) MKL-DNN v2.2.3 (Git Hash 7336ca9f055cf1bfa13efb658fe15dc9b41f0740)\n  - OpenMP 201511 (a.k.a. OpenMP 4.5)\n  - LAPACK is enabled (usually provided by MKL)\n  - NNPACK is enabled\n  - CPU capability usage: AVX2\n  - CUDA Runtime 11.3\n  - NVCC architecture flags: -gencode;arch=compute_37,code=sm_37;-gencode;arch=compute_50,code=sm_50;-gencode;arch=compute_60,code=sm_60;-gencode;arch=compute_70,code=sm_70;-gencode;arch=compute_75,code=sm_75;-gencode;arch=compute_80,code=sm_80;-gencode;arch=compute_86,code=sm_86\n  - CuDNN 8.2\n  - Magma 2.5.2\n  - Build settings: BLAS_INFO=mkl, BUILD_TYPE=Release, CUDA_VERSION=11.3, CUDNN_VERSION=8.2.0, CXX_COMPILER=/opt/rh/devtoolset-7/root/usr/bin/c++, CXX_FLAGS= -Wno-deprecated -fvisibility-inlines-hidden -DUSE_PTHREADPOOL -fopenmp -DNDEBUG -DUSE_KINETO -DUSE_FBGEMM -DUSE_QNNPACK -DUSE_PYTORCH_QNNPACK -DUSE_XNNPACK -DSYMBOLICATE_MOBILE_DEBUG_HANDLE -DEDGE_PROFILER_USE_KINETO -O2 -fPIC -Wno-narrowing -Wall -Wextra -Werror=return-type -Wno-missing-field-initializers -Wno-type-limits -Wno-array-bounds -Wno-unknown-pragmas -Wno-sign-compare -Wno-unused-parameter -Wno-unused-variable -Wno-unused-function -Wno-unused-result -Wno-unused-local-typedefs -Wno-strict-overflow -Wno-strict-aliasing -Wno-error=deprecated-declarations -Wno-stringop-overflow -Wno-psabi -Wno-error=pedantic -Wno-error=redundant-decls -Wno-error=old-style-cast -fdiagnostics-color=always -faligned-new -Wno-unused-but-set-variable -Wno-maybe-uninitialized -fno-math-errno -fno-trapping-math -Werror=format -Wno-stringop-overflow, LAPACK_INFO=mkl, PERF_WITH_AVX=1, PERF_WITH_AVX2=1, PERF_WITH_AVX512=1, TORCH_VERSION=1.10.0, USE_CUDA=ON, USE_CUDNN=ON, USE_EXCEPTION_PTR=1, USE_GFLAGS=OFF, USE_GLOG=OFF, USE_MKL=ON, USE_MKLDNN=ON, USE_MPI=OFF, USE_NCCL=ON, USE_NNPACK=ON, USE_OPENMP=ON, \n'),
             ('TorchVision', '0.11.1+cu113'),
             ('OpenCV', '4.5.4'),
             ('MMEngine', '0.7.3'),
             ('MMCV', '2.0.0'),
             ('MMCV Compiler', 'GCC 9.3'),
             ('MMCV CUDA Compiler', '11.3')])

**Comment by Daniel-deng-yi** [comment]
我也是这个问题
06/02 03:14:23 - mmengine - INFO - Epoch(train)   [1][350/773]  lr: 7.5248e-04  eta: 1 day, 2:28:56  time: 0.2190  data_time: 0.0009  memory: 5479  grad_norm: 0.1394  loss: 0.0033  loss_cls: 0.0033  loss_bbox: 0.0000  loss_dfl: 0.0000
06/02 03:14:34 - mmengine - INFO - Epoch(train)   [1][400/773]  lr: 8.6028e-04  eta: 1 day, 2:06:39  time: 0.2194  data_time: 0.0010  memory: 5479  grad_norm: 0.1089  loss: 0.0016  loss_cls: 0.0016  loss_bbox: 0.0000  loss_dfl: 0.0000
06/02 03:14:45 - mmengine - INFO - Epoch(train)   [1][450/773]  lr: 9.6809e-04  eta: 1 day, 1:48:51  time: 0.2188  data_time: 0.0010  memory: 5479  grad_norm: 0.0879  loss: 0.0008  loss_cls: 0.0008  loss_bbox: 0.0000  loss_dfl: 0.0000
06/02 03:14:56 - mmengine - INFO - Epoch(train)   [1][500/773]  lr: 1.0759e-03  eta: 1 day, 1:34:58  time: 0.2195  data_time: 0.0010  memory: 5479  grad_norm: 0.0711  loss: 0.0004  loss_cls: 0.0004  loss_bbox: 0.0000  loss_dfl: 0.0000
06/02 03:15:07 - mmengine - INFO - Epoch(train)   [1][550/773]  lr: 1.1837e-03  eta: 1 day, 1:23:25  time: 0.2192  data_time: 0.0010  memory: 5479  grad_norm: 0.0586  loss: 0.0003  loss_cls: 0.0003  loss_bbox: 0.0000  

有没有定位到解决方法

**Comment by RDXiaoLu** [comment]
> 我也是这个问题 06/02 03:14:23 - mmengine - INFO - Epoch(train) [1][350/773] lr: 7.5248e-04 eta: 1 day, 2:28:56 time: 0.2190 data_time: 0.0009 memory: 5479 grad_norm: 0.1394 loss: 0.0033 loss_cls: 0.0033 loss_bbox: 0.0000 loss_dfl: 0.0000 06/02 03:14:34 - mmengine - INFO - Epoch(train) [1][400/773] lr: 8.6028e-04 eta: 1 day, 2:06:39 time: 0.2194 data_time: 0.0010 memory: 5479 grad_norm: 0.1089 loss: 0.0016 loss_cls: 0.0016 loss_bbox: 0.0000 loss_dfl: 0.0000 06/02 03:14:45 - mmengine - INFO - Epoch(train) [1][450/773] lr: 9.6809e-04 eta: 1 day, 1:48:51 time: 0.2188 data_time: 0.0010 memory: 5479 grad_norm: 0.0879 loss: 0.0008 loss_cls: 0.0008 loss_bbox: 0.0000 loss_dfl: 0.0000 06/02 03:14:56 - mmengine - INFO - Epoch(train) [1][500/773] lr: 1.0759e-03 eta: 1 day, 1:34:58 time: 0.2195 data_time: 0.0010 memory: 5479 grad_norm: 0.0711 loss: 0.0004 loss_cls: 0.0004 loss_bbox: 0.0000 loss_dfl: 0.0000 06/02 03:15:07 - mmengine - INFO - Epoch(train) [1][550/773] lr: 1.1837e-03 eta: 1 day, 1:23:25 time: 0.2192 data_time: 0.0010 memory: 5479 grad_norm: 0.0586 loss: 0.0003 loss_cls: 0.0003 loss_bbox: 0.0000
> 
> 有没有定位到解决方法

没找到，我也不知道到底哪里除了问题。请问你找到了么。

**Comment by RDXiaoLu** [comment]
have solved.   look https://github.com/open-mmlab/mmyolo/issues/447。  mainly modify train_dataloader and val_dataloader add metainfo....

**Comment by Latitude9527** [comment]
> 

hey ，bro，the context you linked   is no longer exists ,can you roughly describe how it was corrected?

**Comment by MingkaiChen** [comment]
@Latitude9527 Just remove the “。“ at the end of the url :)

https://github.com/open-mmlab/mmyolo/issues/447

**Comment by Latitude9527** [comment]
> @Latitude9527 Just remove the “。“ at the end of the url :)
> 
> #447

ok,thanks

**Comment by Latitude9527** [comment]
> have solved. look [https://github.com/open-mmlab/mmyolo/issues/447。](https://github.com/open-mmlab/mmyolo/issues/447%E3%80%82) mainly modify train_dataloader and val_dataloader add metainfo....

thanks,bro
